### PR TITLE
Add equivalent test coverage for all RHS join impls

### PR DESCRIPTION
**Pull request by maytasm**
Add equivalent test coverage for all RHS join impls

### Description

This PR adds unit tests to make sure that we have equal coverage for all RHS join impls. 
We currently have 2 implementations of "joins", join against a lookup and join against an indexed table. We already have equal coverage tests for LookupJoinable and IndexedTableJoinable via LookupJoinableTest and IndexedTableJoinableTest (added from PR https://github.com/apache/druid/pull/9760). This PR adds the remaining missing tests for LookupJoinMatcher and IndexedTableJoinMatcher (so that we have equal coverage for LookupJoinMatcher and IndexedTableJoinMatcher too).    

This PR has:
- [x] been self-reviewed.
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/licenses.yaml)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [x] added unit tests or modified existing tests to cover new code paths.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

File: processing/src/test/java/org/apache/druid/segment/join/table/LookupJoinMatcherTest.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    Mockito.doReturn(true).when(extractor).canIterate();
+    Mockito.doReturn(lookupMap.entrySet()).when(extractor).iterable();
+  }
+
+  @Test
+  public void testCreateConditionAlwaysFalse()
+  {
+    JoinConditionAnalysis condition = JoinConditionAnalysis.forExpression("0", PREFIX, ExprMacroTable.nil());
+    target = LookupJoinMatcher.create(extractor, leftSelectorFactory, condition, false);
+    target = LookupJoinMatcher.create(extractor, leftSelectorFactory, condition, false);
```
Comment (ccaominh) [inline processing/src/test/java/org/apache/druid/segment/join/table/LookupJoinMatcherTest.java:74]: Should this have `remainderNeeded` set to `true` like it's done in `testCreateConditionAlwaysTrue()`?

File: processing/src/test/java/org/apache/druid/segment/join/table/LookupJoinMatcherTest.java [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+  @Before
+  public void setUp()
+  {
+    Mockito.doReturn(true).when(extractor).canIterate();
+    Mockito.doReturn(lookupMap.entrySet()).when(extractor).iterable();
+  }
+
+  @Test
+  public void testCreateConditionAlwaysFalse()
```
Comment (ccaominh) [inline processing/src/test/java/org/apache/druid/segment/join/table/LookupJoinMatcherTest.java:70]: Is the idea that this should not throw an exception? If so, perhaps add that to the test name or add a comment.

File: processing/src/test/java/org/apache/druid/segment/join/table/LookupJoinMatcherTest.java (hunk 1 above)
Comment (maytasm) [reply processing/src/test/java/org/apache/druid/segment/join/table/LookupJoinMatcherTest.java]: Yes. Typo

File: processing/src/test/java/org/apache/druid/segment/join/table/LookupJoinMatcherTest.java (hunk 2 above)
Comment (maytasm) [reply processing/src/test/java/org/apache/druid/segment/join/table/LookupJoinMatcherTest.java]: Done

**Pull request merged**